### PR TITLE
AnimationUtils: Remove `arraySlice()`.

### PR DESCRIPTION
**Pull request by ycw**
this PR simpilies the polyfill of typedarray.prototype.slice a bit ... in case [its still relevant(?)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray/slice#browser_compatibility)

### CODE REVIEW

File: src/animation/AnimationUtils.js [hunk 1]
```diff
@@ -2,13 +2,13 @@ import { Quaternion } from '../math/Quaternion.js';
...
 
 // same as Array.prototype.slice, but also works on typed arrays
-function arraySlice( array, from, to ) {
+function arraySlice( array, from, to = array.length ) {
 
 	if ( isTypedArray( array ) ) {
 
 		// in ios9 array.subarray(from, undefined) will return empty array
 		// but array.subarray(from) or array.subarray(from, len) is correct
-		return new array.constructor( array.subarray( from, to !== undefined ? to : array.length ) );
```
Comment (Mugen87) [inline src/animation/AnimationUtils.js:11]: I think it's best here to check if latest iOS version handle `array.subarray(from, undefined)` correctly. If so, we can remove the default parameter.

According to my understanding, this was only an issue with WebKit on iOS9 but I suggest to double-check to be sure.

Comment (ycw) [reply src/animation/AnimationUtils.js:11]: i meant if *patching the TypedArray.prototype.slice* is now not revelant... then its safe to remove `arraySlice` from codebase :D

Comment (Mugen87) [reply src/animation/AnimationUtils.js:11]: Compatibility table: https://caniuse.com/mdn-javascript_builtins_typedarray_slice

It looks safe to remove the helper. If for some reasons somebody has issue with a removal, apps can still polyfill `TypedArray.slice()`.

**Review by Mugen87 — APPROVED** [review]

**Comment by Mugen87** [comment]
Do you mind removing `arraySlice()` from the documentation pages as well? There is also a placeholder in the unit test file that can be removed as well.

**Comment by ycw** [comment]
done.

**Pull request merged**